Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabSystemComponent.h

```diff
@@ -237,6 +252,22 @@ namespace AzToolsFramework
         private:
             AZ_DISABLE_COPY_MOVE(PrefabSystemComponent);
 
+            /**
+            * Builds a new Prefab Template out of entities and instances and returns the first instance comprised of
+            * these entities and instances.
+            * @param entities A vector of entities that will be used in the new instance. May be empty.
+            * @param instances A vector of Prefab Instances that will be nested in the new instance, will be consumed and moved.
+            *                  May be empty.
+            * @param filePath The path to associate the template of the new instance to.
+            * @param instance A pointer to the newly created instance which needs initiation.
+            * @param shouldCreateLinks The flag indicating if links should be created between the templates of the instance
+            *        and its nested instances.
+            * @return A pointer to the newly created instance. nullptr on failure.
+            */
+            AZStd::unique_ptr<Instance> CreatePrefab(const AZStd::vector<AZ::Entity*>& entities,
```

Comment:
The way that this function is taking a unique pointer as function argument and returning the same pointer after modifying it is a bit odd. Since this is a private function, you can make this return a void and pass in a reference to the unique pointer as a parameter. That would make this easier to comprehend.